diff --git a/drivers/input/touchscreen/ft5x_ts.c b/drivers/input/touchscreen/ft5x_ts.c
index 5d8dc97..f87e67e 100644
--- a/drivers/input/touchscreen/ft5x_ts.c
+++ b/drivers/input/touchscreen/ft5x_ts.c
@@ -1756,13 +1756,16 @@ ft5x_ts_probe(struct i2c_client *client, const struct i2c_device_id *id)
 	ft5x_ts->early_suspend.resume	= ft5x_ts_resume;
 	register_early_suspend(&ft5x_ts->early_suspend);
 #endif
+	ft5x_set_reg(0x6e, 0x0a);//int enable,low,indicate mode
+	ft5x_set_reg(0x67, 100); //set x sensitivity
+	ft5x_set_reg(0x68, 100); //set y sensitivity
 
 	err = ctp_ops.set_irq_mode("ctp_para", "ctp_int_port", CTP_IRQ_MODE);
 	if(0 != err){
 		pr_info("%s:ctp_ops.set_irq_mode err.\n", __func__);
 		goto exit_set_irq_mode;
 	}
-	err = request_irq(SW_INT_IRQNO_PIO, ft5x_ts_interrupt, IRQF_TRIGGER_FALLING | IRQF_SHARED, "ft5x_ts", ft5x_ts);
+	err = request_irq(SW_INT_IRQNO_PIO, ft5x_ts_interrupt, IRQF_SHARED, "ft5x_ts", ft5x_ts);
 
 	if (err < 0) {
 		dev_err(&client->dev, "ft5x_ts_probe: request irq failed\n");
